rust/parser/test/mod.rs

```diff
@@ -271,7 +271,10 @@ $x has release-date < 1986-03-03T00:00,
     let expected = typeql_match!(var("x")
         .has((
             "release-date",
-            lt(NaiveDateTime::new(NaiveDate::from_ymd(1986, 3, 3), NaiveTime::from_hms(0, 0, 0)))
+            lt(NaiveDateTime::new(
+                NaiveDate::from_ymd_opt(1986, 3, 3).unwrap(),
+                NaiveTime::from_hms_opt(0, 0, 0).unwrap()
+            ))
```

Comment:
`from_ymd`, `from_hms`, and `from_hms_millis` are all deprecated, and their new `_opt` variants (which validate the input and return an `Option<T>`) are to be used instead.